rasa/nlu/featurizers/dense_featurizer/convert_featurizer.py

```diff
@@ -72,19 +70,93 @@ def _compute_features(
+
+        if not self.return_sequence:
+            return sentence_encodings
+
+        return self._compute_sequence_encodings(
+            batch_examples, sentence_encodings, attribute
+        )
+
+    def _compute_sentence_encodings(
+        self, batch_examples: List[Message], attribute: Text = TEXT_ATTRIBUTE
+    ) -> np.ndarray:
         # Get text for attribute of each example
         batch_attribute_text = [ex.get(attribute) for ex in batch_examples]
+        sentence_encodings = self._sentence_encoding_of_text(batch_attribute_text)
 
-        batch_features = self._run_model_on_text(batch_attribute_text)
+        # convert them to a sequence of 1
+        return np.reshape(sentence_encodings, (len(batch_examples), 1, -1))
 
-        return batch_features
+    def _compute_sequence_encodings(
+        self,
+        batch_examples: List[Message],
+        sentence_encodings: np.ndarray,
+        attribute: Text = TEXT_ATTRIBUTE,
+    ) -> np.ndarray:
+        list_of_tokens = [
+            example.get(TOKENS_NAMES[attribute]) for example in batch_examples
+        ]
+
+        cls_token_used = list_of_tokens[0][-1] == CLS_TOKEN
+
+        number_of_tokens_in_sentence = [
+            len(sent_tokens) for sent_tokens in list_of_tokens
+        ]
+
+        # join the tokens to get a clean text to ensure the sequence length of
+        # the returned embeddings from ConveRT matches the length of the tokens
+        tokenized_texts = self._tokens_to_text(list_of_tokens)
+        sequence_encodings = self._sequence_encoding_of_text(tokenized_texts)
+
+        if not cls_token_used:
+            return sequence_encodings
+
+        final_embeddings = []
+        for index in range(len(batch_examples)):
+            sequence_length = number_of_tokens_in_sentence[index]
+            sequence_encoding = sequence_encodings[index][:sequence_length]
+            sentence_encoding = sentence_encodings[index]
+
+            # tile sequence encoding to duplicate as sentence encodings have size
+            # 1024 and sequence encodings only have a dimensionality of 512
+            sequence_encoding = np.tile(sequence_encoding, (1, 2))
+            # add sentence encoding to the end (position of cls token)
+            sequence_encoding = np.concatenate(
+                [sequence_encoding, sentence_encoding], axis=0
+            )
+
+            final_embeddings.append(sequence_encoding)
+
+        return np.array(final_embeddings)
 
-    def _run_model_on_text(self, batch: List[Text]) -> np.ndarray:
+    @staticmethod
+    def _tokens_to_text(list_of_tokens: List[List[Token]]) -> List[Text]:
+        texts = []
+        for sent_tokens in list_of_tokens:
+            text = ""
+            offset = 0
+            for token in sent_tokens:
+                if offset != token.start:
+                    text += " "
+                text += token.text.replace("﹏", "")
```

Comment:
Would it be possible to perform this post-processing on tokens inside the tokenizer itself?
Would it mess up the entity offset annotations?